.github/workflows/moodle-plugin-ci.yml

```diff
@@ -0,0 +1,119 @@
+        image: postgres:9.6
+        env:
+          POSTGRES_USER: 'postgres'
+          POSTGRES_HOST_AUTH_METHOD: 'trust'
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 3
+      mariadb:
+        image: mariadb:10
+        env:
+          MYSQL_USER: 'root'
+          MYSQL_ALLOW_EMPTY_PASSWORD: "true"
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval 10s --health-timeout 5s --health-retries 3
+
+    # PHP Compatibility Matrix Moodle           End of life
+    # Moodle 3.8: php7.1, php7.2, php7.3        10 May 2021
+    # Moodle 3.9: php7.2, php7.3, php7.4        8 May 2023
+    # Moodle 3.10: php7.2, php7.3, php7.4       9 May 2022
+    # earlier versions of Moodle and php are not included, as support was dropped
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Overall strategy is to run as few jobs as possible, but:
+        # - each php version at least once
+        # - each moodle version at least once
+        # - each database at least once
+        include:
+          # php edge cases with pgsql
+          - {php: '7.1', moodle-branch: MOODLE_38_STABLE, database: pgsql}
+          - {php: '7.2', moodle-branch: MOODLE_39_STABLE, database: mariadb}
+          - {php: '7.3', moodle-branch: MOODLE_39_STABLE, database: pgsql}
+          - {php: '7.4', moodle-branch: MOODLE_310_STABLE, database: mariadb}
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          path: plugin
+
+      - name: Setup PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          coverage: none
+
+      - name: Initialise moodle-plugin-ci
+        run: |
+          composer create-project -n --no-dev --prefer-dist moodlehq/moodle-plugin-ci ci ^3
+          echo $(cd ci/bin; pwd) >> $GITHUB_PATH
+          echo $(cd ci/vendor/bin; pwd) >> $GITHUB_PATH
+          sudo locale-gen en_AU.UTF-8
+          echo "NVM_DIR=$HOME/.nvm" >> $GITHUB_ENV
+
+      - name: Install moodle-plugin-ci
+        run: |
+          moodle-plugin-ci install --plugin ./plugin --db-host=127.0.0.1
+        env:
+          DB: ${{ matrix.database }}
+          MOODLE_BRANCH: ${{ matrix.moodle-branch }}
+
+      - name: PHP Lint
+        if: ${{ always() }}
+        run: moodle-plugin-ci phplint
+
+      - name: PHP Copy/Paste Detector
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpcpd
+
+      - name: PHP Mess Detector
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpmd
+
+      - name: Moodle Code Checker
+        if: ${{ always() }}
+        run: moodle-plugin-ci codechecker --max-warnings 0
+        continue-on-error: true
+
+      - name: Moodle PHPDoc Checker
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpdoc
+        continue-on-error: true
```

Comment:
Anyone an idea how to correctly write inline phpdocs tag @link to a class or function? E.g. https://github.com/frankkoch/moodle-mod_studentquiz/blob/master/lib.php#L300

If solved, we can omit `continue-on-error`

Reply:
I think you need to use {@see ...} here, instead of {@link ...}

Reply:
I've compiled the affected CI output below:

```
/home/runner/work/moodle-mod_studentquiz/moodle-mod_studentquiz/moodle/mod/studentquiz/backup/moodle2/restore_studentquiz_stepslib.php
    Line 49: Inline phpdocs tag {@link restore_path_element} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}

/home/runner/work/moodle-mod_studentquiz/moodle-mod_studentquiz/moodle/mod/studentquiz/backup/moodle2/restore_studentquiz_activity_task.class.php
    Line 84: Inline phpdocs tag {@link restore_logs_processor} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 86: Inline phpdocs tag {@link restore_log_rule} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 100: Inline phpdocs tag {@link restore_logs_processor} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 102: Inline phpdocs tag {@link restore_log_rule} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}

/home/runner/work/moodle-mod_studentquiz/moodle-mod_studentquiz/moodle/mod/studentquiz/lib.php
    Line 45: Inline phpdocs tag {@link plugin_supports()} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 279: Inline phpdocs tag {@link studentquiz_print_recent_mod_activity()} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 295: Inline phpdocs tag {@link studentquiz_get_recent_mod_activity()} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 300: Inline phpdocs tag {@link get_module_types_names()} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 340: Inline phpdocs tag {@link file_browser::get_file_info_context_module()} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 410: Inline phpdocs tag {@link questionlib.php} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}

/home/runner/work/moodle-mod_studentquiz/moodle-mod_studentquiz/moodle/mod/studentquiz/locallib.php
    Line 467: Inline phpdocs tag {@link message_send()} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 504: Inline phpdocs tag {@link mod_studentquiz_send_notification} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
    Line 511: Inline phpdocs tag {@link message_send()} with incorrect contents found. It must match {@link valid URL} or {@see valid FQSEN}
```

We could also omit @link, but I believe I've seen those in Moodle core as well. At least we should try to improve it anyway, that's why I've opened this.